docroot/modules/custom/mass_content/src/ImageStylesWarmer.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Drupal\mass_content;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Image\ImageFactory;
+use Drupal\file\FileInterface;
+
+/**
+ * Defines an images styles warmer.
+ */
+class ImageStylesWarmer {
+
+  /**
+   * The file entity storage.
+   *
+   * @var \Drupal\file\FileStorageInterface
+   */
+  protected $file;
+
+  /**
+   * The image factory.
+   *
+   * @var \Drupal\Core\Image\ImageFactory
+   */
+  protected $image;
+
+  /**
+   * The image style entity storage.
+   *
+   * @var \Drupal\image\ImageStyleStorageInterface
+   */
+  protected $imageStyles;
+
+  /**
+   * Constructs a ImageStylesWarmer object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $file_storage
+   *   The file storage.
+   * @param \Drupal\Core\Image\ImageFactory $image_factory
+   *   The image factory.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $image_style_storage
+   *   The image style storage.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function __construct(EntityTypeManagerInterface $file_storage, ImageFactory $image_factory, EntityTypeManagerInterface $image_style_storage) {
+    $this->file = $file_storage->getStorage('file');
+    $this->image = $image_factory;
+    $this->imageStyles = $image_style_storage->getStorage('image_style');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function warmUp(FileInterface $file) {
+    $initialImageStyles = ['action_banner_large_focal_point', 'hero1600x400_fp'];
+    $this->doWarmUp($file, $initialImageStyles);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function doWarmUp(FileInterface $file, array $image_styles) {
+    if (empty($image_styles) || !$this->validateImage($file)) {
```

Comment:
How about we reverse the order of these conditionals, to match the method arguments? "First validate the file, then check the image styles".

```suggestion
    if (!$this->validateImage($file) || empty($image_styles)) {
```

Reply:
What's the difference?

Reply:
I was mainly thinking about this from a "reading code" perspective. The method is "file, image styles", and it's generally logical to validate parameters in the same order they are in the method signature.

_however_ there are times it's good to break from that, such as if `validateImage()` is much slower, so by checking other conditions you can be a tad faster. My guess is it doesn't matter much in this case performance wise.